diff -Naur a/masterpdfeditor5.desktop b/masterpdfeditor5.desktop
--- a/masterpdfeditor5.desktop	2026-02-18 10:03:12.000000000 +0100
+++ b/masterpdfeditor5.desktop	2026-02-20 18:58:26.744385626 +0100
@@ -1,10 +1,10 @@
 [Desktop Entry]
 Name=Master PDF Editor 5
 Comment=Edit PDF files
-Exec=/opt/master-pdf-editor-5/masterpdfeditor5 %f
-Path=/opt/master-pdf-editor-5
+Exec=/usr/lib/master-pdf-editor-5/masterpdfeditor5 %f
+Path=/usr/lib/master-pdf-editor-5
 Terminal=false
-Icon=/opt/master-pdf-editor-5/masterpdfeditor5.png
+Icon=masterpdfeditor5
 Type=Application
 Categories=Office;Graphics;
 MimeType=application/pdf;application/x-bzpdf;application/x-gzpdf;
