diff -Naur a/config_host.mk.in b/config_host.mk.in
--- a/config_host.mk.in	2025-10-22 14:04:25.000000000 +0200
+++ b/config_host.mk.in	2025-11-02 12:08:55.085735958 +0100
@@ -48,7 +48,6 @@
 export BOOST_FILESYSTEM_LIB=@BOOST_FILESYSTEM_LIB@
 export BOOST_IOSTREAMS_LIB=@BOOST_IOSTREAMS_LIB@
 export BOOST_LDFLAGS=@BOOST_LDFLAGS@
-export BOOST_SYSTEM_LIB=@BOOST_SYSTEM_LIB@
 export BOX2D_CFLAGS=@BOX2D_CFLAGS@
 export BOX2D_LIBS=@BOX2D_LIBS@
 export BRAND_INTRO_IMAGES=@BRAND_INTRO_IMAGES@
diff -Naur a/configure.ac b/configure.ac
--- a/configure.ac	2025-10-22 14:04:25.000000000 +0200
+++ b/configure.ac	2025-11-02 12:10:52.462584663 +0100
@@ -11453,19 +11453,9 @@
 dnl ===================================================================
 dnl Orcus
 dnl ===================================================================
-libo_CHECK_SYSTEM_MODULE([orcus],[ORCUS],[liborcus-0.20 >= 0.20.0])
-if test "$with_system_orcus" != "yes"; then
-    if test "$SYSTEM_BOOST" = "TRUE"; then
-        dnl Link with Boost.System
-        dnl This seems to be necessary since boost 1.50 (1.48 does not need it,
-        dnl 1.49 is untested). The macro BOOST_THREAD_DONT_USE_SYSTEM mentioned
-        dnl in documentation has no effect.
-        AX_BOOST_SYSTEM
-    fi
-fi
+libo_CHECK_SYSTEM_MODULE([orcus],[ORCUS],[liborcus-0.21 >= 0.21.0])
 dnl FIXME by renaming SYSTEM_LIBORCUS to SYSTEM_ORCUS in the build system world
 SYSTEM_LIBORCUS=$SYSTEM_ORCUS
-AC_SUBST([BOOST_SYSTEM_LIB])
 AC_SUBST(SYSTEM_LIBORCUS)
 
 dnl ===================================================================
diff -Naur a/download.lst b/download.lst
--- a/download.lst	2025-10-22 14:04:25.000000000 +0200
+++ b/download.lst	2025-11-02 12:08:55.089735987 +0100
@@ -14,8 +14,8 @@
 # so that git cherry-pick
 # will not run into conflicts
 # please repack the tarball using external/boost/repack_tarball.sh
-BOOST_SHA256SUM := 5c67a448c562f1606e38203ba7ed7e8d7453581b6b9ca324e96205eae0da5ff8
-BOOST_TARBALL := boost_1_88_0.tar.xz
+BOOST_SHA256SUM := b3a976c659961f1d7fae9a8e58d6729a11f5473abaf2ef0e623f53777c86475d
+BOOST_TARBALL := boost_1_89_0.tar.xz
 # three static lines
 # so that git cherry-pick
 # will not run into conflicts
@@ -554,8 +554,8 @@
 # three static lines
 # so that git cherry-pick
 # will not run into conflicts
-ORCUS_SHA256SUM := ec27f30e8445a2a3f307f7e829fc446fd48193150b7f8f23bb5bfb25ec6e4e27
-ORCUS_TARBALL := liborcus-0.20.1.tar.xz
+ORCUS_SHA256SUM := 1c6e473d6b8a63d61d0e6874a1762d6f0e0e78338d763e48ac93126ecde19b37
+ORCUS_TARBALL := liborcus-0.21.0.tar.xz
 # three static lines
 # so that git cherry-pick
 # will not run into conflicts
diff -Naur a/external/boost/Module_boost.mk b/external/boost/Module_boost.mk
--- a/external/boost/Module_boost.mk	2025-10-22 14:04:25.000000000 +0200
+++ b/external/boost/Module_boost.mk	2025-11-02 12:08:55.089735987 +0100
@@ -13,7 +13,6 @@
 	StaticLibrary_boost_date_time \
 	StaticLibrary_boost_filesystem \
 	StaticLibrary_boost_locale \
-	StaticLibrary_boost_system \
 	StaticLibrary_boost_iostreams \
 	UnpackedTarball_boost \
 ))
diff -Naur a/external/boost/StaticLibrary_boost_system.mk b/external/boost/StaticLibrary_boost_system.mk
--- a/external/boost/StaticLibrary_boost_system.mk	2025-10-22 14:04:25.000000000 +0200
+++ b/external/boost/StaticLibrary_boost_system.mk	1970-01-01 01:00:00.000000000 +0100
@@ -1,29 +0,0 @@
-# -*- Mode: makefile-gmake; tab-width: 4; indent-tabs-mode: t -*-
-#
-# This file is part of the LibreOffice project.
-#
-# This Source Code Form is subject to the terms of the Mozilla Public
-# License, v. 2.0. If a copy of the MPL was not distributed with this
-# file, You can obtain one at http://mozilla.org/MPL/2.0/.
-#
-
-$(eval $(call gb_StaticLibrary_StaticLibrary,boost_system))
-
-$(eval $(call gb_StaticLibrary_use_unpacked,boost_system,boost))
-
-$(eval $(call gb_StaticLibrary_set_warnings_disabled,boost_system))
-
-# disable "auto link" "feature" on MSVC
-$(eval $(call gb_StaticLibrary_add_defs,boost_system,\
-	-DBOOST_ALL_NO_LIB \
-))
-
-$(eval $(call gb_StaticLibrary_use_external,boost_system,boost_headers))
-
-$(eval $(call gb_StaticLibrary_set_generated_cxx_suffix,boost_system,cpp))
-
-$(eval $(call gb_StaticLibrary_add_generated_exception_objects,boost_system,\
-	UnpackedTarball/boost/libs/system/src/error_code \
-))
-
-# vim: set noet sw=4 ts=4:
diff -Naur a/external/liborcus/ExternalPackage_liborcus.mk b/external/liborcus/ExternalPackage_liborcus.mk
--- a/external/liborcus/ExternalPackage_liborcus.mk	2025-10-22 14:04:25.000000000 +0200
+++ b/external/liborcus/ExternalPackage_liborcus.mk	2025-11-02 12:08:55.089735987 +0100
@@ -12,11 +12,11 @@
 $(eval $(call gb_ExternalPackage_use_external_project,liborcus,liborcus))
 
 ifeq ($(OS),MACOSX)
-$(eval $(call gb_ExternalPackage_add_file,liborcus,$(LIBO_LIB_FOLDER)/liborcus-0.20.0.dylib,src/liborcus/.libs/liborcus-0.20.0.dylib))
-$(eval $(call gb_ExternalPackage_add_file,liborcus,$(LIBO_LIB_FOLDER)/liborcus-parser-0.20.0.dylib,src/parser/.libs/liborcus-parser-0.20.0.dylib))
+$(eval $(call gb_ExternalPackage_add_file,liborcus,$(LIBO_LIB_FOLDER)/liborcus-0.21.0.dylib,src/liborcus/.libs/liborcus-0.21.0.dylib))
+$(eval $(call gb_ExternalPackage_add_file,liborcus,$(LIBO_LIB_FOLDER)/liborcus-parser-0.21.0.dylib,src/parser/.libs/liborcus-parser-0.21.0.dylib))
 else ifeq ($(DISABLE_DYNLOADING),)
-$(eval $(call gb_ExternalPackage_add_file,liborcus,$(LIBO_LIB_FOLDER)/liborcus-0.20.so.0,src/liborcus/.libs/liborcus-0.20.so.0.0.0))
-$(eval $(call gb_ExternalPackage_add_file,liborcus,$(LIBO_LIB_FOLDER)/liborcus-parser-0.20.so.0,src/parser/.libs/liborcus-parser-0.20.so.0.0.0))
+$(eval $(call gb_ExternalPackage_add_file,liborcus,$(LIBO_LIB_FOLDER)/liborcus-0.21.so.0,src/liborcus/.libs/liborcus-0.21.so.0.0.0))
+$(eval $(call gb_ExternalPackage_add_file,liborcus,$(LIBO_LIB_FOLDER)/liborcus-parser-0.21.so.0,src/parser/.libs/liborcus-parser-0.21.so.0.0.0))
 endif
 
 # vim: set noet sw=4 ts=4:
diff -Naur a/external/liborcus/ExternalProject_liborcus.mk b/external/liborcus/ExternalProject_liborcus.mk
--- a/external/liborcus/ExternalProject_liborcus.mk	2025-10-22 14:04:25.000000000 +0200
+++ b/external/liborcus/ExternalProject_liborcus.mk	2025-11-02 12:08:55.093736016 +0100
@@ -15,7 +15,6 @@
     boost_headers \
     boost_filesystem \
     boost_iostreams \
-    boost_system \
 	mdds_headers \
     zlib \
 ))
@@ -39,9 +38,9 @@
 liborcus_LIBS+=-lz
 endif
 ifneq ($(SYSTEM_BOOST),)
-liborcus_LIBS+=$(BOOST_SYSTEM_LIB) $(BOOST_IOSTREAMS_LIB) $(BOOST_FILESYSTEM_LIB)
+liborcus_LIBS+=$(BOOST_IOSTREAMS_LIB) $(BOOST_FILESYSTEM_LIB)
 else
-liborcus_LIBS+=-L$(gb_StaticLibrary_WORKDIR) -lboost_system -lboost_iostreams -lboost_filesystem
+liborcus_LIBS+=-L$(gb_StaticLibrary_WORKDIR) -lboost_iostreams -lboost_filesystem
 endif
 ifeq ($(OS),ANDROID)
 liborcus_LIBS+=$(gb_STDLIBS)
@@ -89,6 +88,7 @@
 		$(if $(liborcus_LDFLAGS),LDFLAGS='$(liborcus_LDFLAGS)') \
 		MDDS_CFLAGS='$(MDDS_CFLAGS)' \
 		MDDS_LIBS=' ' \
+		rm -f configure && ./autogen.sh \
 		MAKE=$(MAKE) $(gb_RUN_CONFIGURE) ./configure \
 			--with-pic \
 			$(if $(DISABLE_DYNLOADING), \
@@ -115,8 +115,8 @@
 		   $(MAKE) \
 		$(if $(filter MACOSX,$(OS)),\
 			&& $(PERL) $(SRCDIR)/solenv/bin/macosx-change-install-names.pl shl OOO \
-				$(EXTERNAL_WORKDIR)/src/liborcus/.libs/liborcus-0.20.0.dylib \
-				$(EXTERNAL_WORKDIR)/src/parser/.libs/liborcus-parser-0.20.0.dylib \
+				$(EXTERNAL_WORKDIR)/src/liborcus/.libs/liborcus-0.21.0.dylib \
+				$(EXTERNAL_WORKDIR)/src/parser/.libs/liborcus-parser-0.21.0.dylib \
 		) \
 	)
 	$(call gb_Trace_EndRange,liborcus,EXTERNAL)
diff -Naur a/external/liborcus/Library_orcus.mk b/external/liborcus/Library_orcus.mk
--- a/external/liborcus/Library_orcus.mk	2025-10-22 14:04:25.000000000 +0200
+++ b/external/liborcus/Library_orcus.mk	2025-11-02 12:08:55.093736016 +0100
@@ -15,7 +15,6 @@
 	boost_headers \
 	boost_filesystem \
 	boost_iostreams \
-	boost_system \
 	mdds_headers \
 	zlib \
 ))
diff -Naur a/external/liborcus/Library_orcus-parser.mk b/external/liborcus/Library_orcus-parser.mk
--- a/external/liborcus/Library_orcus-parser.mk	2025-10-22 14:04:25.000000000 +0200
+++ b/external/liborcus/Library_orcus-parser.mk	2025-11-02 12:08:55.093736016 +0100
@@ -14,7 +14,6 @@
 $(eval $(call gb_Library_use_externals,orcus-parser,\
 	boost_headers \
 	boost_filesystem \
-	boost_system \
 	mdds_headers \
 	zlib \
 ))
diff -Naur a/external/liborcus/UnpackedTarball_liborcus.mk b/external/liborcus/UnpackedTarball_liborcus.mk
--- a/external/liborcus/UnpackedTarball_liborcus.mk	2025-10-22 14:04:25.000000000 +0200
+++ b/external/liborcus/UnpackedTarball_liborcus.mk	2025-11-02 12:08:55.093736016 +0100
@@ -15,7 +15,6 @@
 
 $(eval $(call gb_UnpackedTarball_add_patches,liborcus,\
 	external/liborcus/rpath.patch.0 \
-	external/liborcus/libtool.patch.0 \
 ))
 
 ifeq ($(OS),WNT)
diff -Naur a/m4/ax_boost_filesystem.m4 b/m4/ax_boost_filesystem.m4
--- a/m4/ax_boost_filesystem.m4	2025-10-22 14:04:25.000000000 +0200
+++ b/m4/ax_boost_filesystem.m4	2025-11-02 12:08:55.093736016 +0100
@@ -63,10 +63,6 @@
 		LDFLAGS="$LDFLAGS $BOOST_LDFLAGS"
 		export LDFLAGS
 
-		LIBS_SAVED=$LIBS
-		LIBS="$LIBS $BOOST_SYSTEM_LIB"
-		export LIBS
-
         AC_CACHE_CHECK(whether the Boost::Filesystem library is available,
 					   ax_cv_boost_filesystem,
         [AC_LANG_PUSH([C++])
@@ -113,6 +109,5 @@
 
 		CPPFLAGS="$CPPFLAGS_SAVED"
 		LDFLAGS="$LDFLAGS_SAVED"
-		LIBS="$LIBS_SAVED"
 	fi
 ])
diff -Naur a/m4/ax_boost_system.m4 b/m4/ax_boost_system.m4
--- a/m4/ax_boost_system.m4	2025-10-22 14:04:25.000000000 +0200
+++ b/m4/ax_boost_system.m4	1970-01-01 01:00:00.000000000 +0100
@@ -1,121 +0,0 @@
-# ===========================================================================
-#     https://www.gnu.org/software/autoconf-archive/ax_boost_system.html
-# ===========================================================================
-#
-# SYNOPSIS
-#
-#   AX_BOOST_SYSTEM
-#
-# DESCRIPTION
-#
-#   Test for System library from the Boost C++ libraries. The macro requires
-#   a preceding call to AX_BOOST_BASE. Further documentation is available at
-#   <http://randspringer.de/boost/index.html>.
-#
-#   This macro calls:
-#
-#     AC_SUBST(BOOST_SYSTEM_LIB)
-#
-#   And sets:
-#
-#     HAVE_BOOST_SYSTEM
-#
-# LICENSE
-#
-#   Copyright (c) 2008 Thomas Porschberg <thomas@randspringer.de>
-#   Copyright (c) 2008 Michael Tindal
-#   Copyright (c) 2008 Daniel Casimiro <dan.casimiro@gmail.com>
-#
-#   Copying and distribution of this file, with or without modification, are
-#   permitted in any medium without royalty provided the copyright notice
-#   and this notice are preserved. This file is offered as-is, without any
-#   warranty.
-
-#serial 20
-
-AC_DEFUN([AX_BOOST_SYSTEM],
-[
-	AC_ARG_WITH([boost-system],
-	AS_HELP_STRING([--with-boost-system@<:@=special-lib@:>@],
-                   [use the System library from boost - it is possible to specify a certain library for the linker
-                        e.g. --with-boost-system=boost_system-gcc-mt ]),
-        [
-        if test "$withval" = "no"; then
-			want_boost="no"
-        elif test "$withval" = "yes"; then
-            want_boost="yes"
-            ax_boost_user_system_lib=""
-        else
-		    want_boost="yes"
-		ax_boost_user_system_lib="$withval"
-		fi
-        ],
-        [want_boost="yes"]
-	)
-
-	if test "x$want_boost" = "xyes"; then
-        AC_REQUIRE([AC_PROG_CC])
-        AC_REQUIRE([AC_CANONICAL_BUILD])
-		CPPFLAGS_SAVED="$CPPFLAGS"
-		CPPFLAGS="$CPPFLAGS $BOOST_CPPFLAGS"
-		export CPPFLAGS
-
-		LDFLAGS_SAVED="$LDFLAGS"
-		LDFLAGS="$LDFLAGS $BOOST_LDFLAGS"
-		export LDFLAGS
-
-        AC_CACHE_CHECK(whether the Boost::System library is available,
-					   ax_cv_boost_system,
-        [AC_LANG_PUSH([C++])
-			 CXXFLAGS_SAVE=$CXXFLAGS
-			 CXXFLAGS=
-
-			 AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[@%:@include <boost/system/error_code.hpp>]],
-				    [[boost::system::error_category *a = 0;]])],
-                   ax_cv_boost_system=yes, ax_cv_boost_system=no)
-			 CXXFLAGS=$CXXFLAGS_SAVE
-             AC_LANG_POP([C++])
-		])
-		if test "x$ax_cv_boost_system" = "xyes"; then
-			AC_SUBST(BOOST_CPPFLAGS)
-
-			AC_DEFINE(HAVE_BOOST_SYSTEM,,[define if the Boost::System library is available])
-            BOOSTLIBDIR=`echo $BOOST_LDFLAGS | sed -e 's/@<:@^\/@:>@*//'`
-
-			LDFLAGS_SAVE=$LDFLAGS
-            if test "x$ax_boost_user_system_lib" = "x"; then
-                for libextension in `ls -r $BOOSTLIBDIR/libboost_system* 2>/dev/null | sed 's,.*/lib,,' | sed 's,\..*,,'` ; do
-                     ax_lib=${libextension}
-				    AC_CHECK_LIB($ax_lib, exit,
-                                 [BOOST_SYSTEM_LIB="-l$ax_lib"; AC_SUBST(BOOST_SYSTEM_LIB) link_system="yes"; break],
-                                 [link_system="no"])
-				done
-                if test "x$link_system" != "xyes"; then
-                for libextension in `ls -r $BOOSTLIBDIR/boost_system* 2>/dev/null | sed 's,.*/,,' | sed -e 's,\..*,,'` ; do
-                     ax_lib=${libextension}
-				    AC_CHECK_LIB($ax_lib, exit,
-                                 [BOOST_SYSTEM_LIB="-l$ax_lib"; AC_SUBST(BOOST_SYSTEM_LIB) link_system="yes"; break],
-                                 [link_system="no"])
-				done
-                fi
-
-            else
-               for ax_lib in $ax_boost_user_system_lib boost_system-$ax_boost_user_system_lib; do
-				      AC_CHECK_LIB($ax_lib, exit,
-                                   [BOOST_SYSTEM_LIB="-l$ax_lib"; AC_SUBST(BOOST_SYSTEM_LIB) link_system="yes"; break],
-                                   [link_system="no"])
-                  done
-
-            fi
-            if test "x$ax_lib" = "x"; then
-                AC_MSG_ERROR(Could not find a version of the Boost::System library!)
-            fi
-			if test "x$link_system" = "xno"; then
-				AC_MSG_ERROR(Could not link against $ax_lib !)
-			fi
-		fi
-
-		CPPFLAGS="$CPPFLAGS_SAVED"
-	LDFLAGS="$LDFLAGS_SAVED"
-	fi
-])
diff -Naur a/RepositoryExternal.mk b/RepositoryExternal.mk
--- a/RepositoryExternal.mk	2025-10-22 14:04:25.000000000 +0200
+++ b/RepositoryExternal.mk	2025-11-02 12:10:26.998400139 +0100
@@ -642,13 +642,6 @@
 
 gb_ExternalProject__use_boost_iostreams :=
 
-define gb_LinkTarget__use_boost_system
-$(call gb_LinkTarget__use_boost_lib,$(1),$(BOOST_SYSTEM_LIB))
-
-endef
-
-gb_ExternalProject__use_boost_system :=
-
 define gb_LinkTarget__use_boost_headers
 $(call gb_LinkTarget_set_include,$(1),\
 	$$(INCLUDE) \
@@ -701,15 +694,6 @@
 $(call gb_ExternalProject_use_static_libraries,$(1),boost_iostreams)
 endef
 
-define gb_LinkTarget__use_boost_system
-$(call gb_LinkTarget__use_boost_lib,$(1),boost_system)
-
-endef
-
-define gb_ExternalProject__use_boost_system
-$(call gb_ExternalProject_use_static_libraries,$(1),boost_system)
-endef
-
 define gb_LinkTarget__use_boost_headers
 $(call gb_LinkTarget_use_unpacked,$(1),boost)
 $(call gb_LinkTarget_set_include,$(1),\
@@ -3398,12 +3382,11 @@
 )
 
 $(call gb_LinkTarget_add_libs,$(1),\
-       -L$(gb_UnpackedTarball_workdir)/liborcus/src/liborcus/.libs -lorcus-0.20 \
+       -L$(gb_UnpackedTarball_workdir)/liborcus/src/liborcus/.libs -lorcus-0.21 \
 )
 
 $(if $(SYSTEM_BOOST), \
 	$(call gb_LinkTarget_add_ldflags,$(1),$(BOOST_LDFLAGS)) \
-	$(call gb_LinkTarget_add_libs,$(1),$(BOOST_SYSTEM_LIB)) \
 )
 
 endef
@@ -3417,7 +3400,7 @@
 )
 
 $(call gb_LinkTarget_add_libs,$(1),\
-	-L$(gb_UnpackedTarball_workdir)/liborcus/src/parser/.libs -lorcus-parser-0.20 \
+	-L$(gb_UnpackedTarball_workdir)/liborcus/src/parser/.libs -lorcus-parser-0.21 \
 )
 
 endef
diff -Naur a/solenv/gbuild/platform/com_GCC_defs.mk b/solenv/gbuild/platform/com_GCC_defs.mk
--- a/solenv/gbuild/platform/com_GCC_defs.mk	2025-10-22 14:04:25.000000000 +0200
+++ b/solenv/gbuild/platform/com_GCC_defs.mk	2025-11-02 12:08:55.093736016 +0100
@@ -46,7 +46,6 @@
 gb_AFLAGS := $(AFLAGS)
 
 gb_COMPILERDEFS := \
-	-DBOOST_SYSTEM_NO_DEPRECATED \
 	-DCPPU_ENV=$(CPPU_ENV) \
 	$(if $(filter EMSCRIPTEN,$(OS)),-U_FORTIFY_SOURCE) \
 
diff -Naur a/solenv/gbuild/platform/com_MSC_defs.mk b/solenv/gbuild/platform/com_MSC_defs.mk
--- a/solenv/gbuild/platform/com_MSC_defs.mk	2025-10-22 14:04:25.000000000 +0200
+++ b/solenv/gbuild/platform/com_MSC_defs.mk	2025-11-02 12:08:55.093736016 +0100
@@ -47,7 +47,6 @@
 
 gb_COMPILERDEFS := \
 	-DBOOST_OPTIONAL_USE_OLD_DEFINITION_OF_NONE \
-	-DBOOST_SYSTEM_NO_DEPRECATED \
 	-D_SILENCE_CXX17_CODECVT_HEADER_DEPRECATION_WARNING \
 	-D_SILENCE_CXX17_OLD_ALLOCATOR_MEMBERS_DEPRECATION_WARNING \
 	-D_SILENCE_CXX17_RESULT_OF_DEPRECATION_WARNING \
diff -Naur a/vcl/commonfuzzer.mk b/vcl/commonfuzzer.mk
--- a/vcl/commonfuzzer.mk	2025-10-22 14:04:25.000000000 +0200
+++ b/vcl/commonfuzzer.mk	2025-11-02 12:08:55.093736016 +0100
@@ -14,7 +14,6 @@
     orcus-parser \
     boost_locale \
     boost_filesystem \
-    boost_system \
     boost_iostreams \
     curl \
     dtoa \
