diff -Naur a/healer/Makefile b/healer/Makefile
--- a/healer/Makefile	2026-07-10 13:06:28.000000000 +0200
+++ b/healer/Makefile	2026-07-17 10:57:29.238683777 +0200
@@ -69,7 +69,7 @@
 
 install-healer: default
 	$(INSTALL) -m 755 -d $(PKG_LIBEXEC_DIR)
-	$(LTINSTALL) -m 755 $(BUILD_TARGETS) $(PKG_LIBEXEC_DIR)
+	$(INSTALL) -m 755 $(BUILD_TARGETS) $(PKG_LIBEXEC_DIR)
 
 install-systemd: default
 	$(INSTALL) -m 755 -d $(SYSTEMD_SYSTEM_UNIT_DIR)
