diff -Naur a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	2025-09-22 14:09:24.000000000 +0200
+++ b/CMakeLists.txt	2025-11-08 22:05:59.581002259 +0100
@@ -314,12 +314,6 @@
 find_program(PODMAN podman)
 mark_as_advanced(PODMAN)
 
-find_program(SCDOC NAMES scdoc REQUIRED)
-mark_as_advanced(SCDOC)
-
-find_program(SCD2HTML NAMES scd2html)
-mark_as_advanced(SCD2HTML)
-
 function(chkdef func req)
 	string(TOUPPER ${func} FUNC)
 	set(HAVENAME HAVE_${FUNC})
