diff -Naur a/Modules/GNUInstallDirs.cmake b/Modules/GNUInstallDirs.cmake
--- a/Modules/GNUInstallDirs.cmake	2025-09-30 16:48:33.000000000 +0200
+++ b/Modules/GNUInstallDirs.cmake	2025-11-05 14:08:44.057948135 +0100
@@ -445,7 +445,7 @@
     elseif(NOT DEFINED system_type)
       # not debian, alpine, arch, or conda so rely on CMAKE_SIZEOF_VOID_P:
       if("${CMAKE_SIZEOF_VOID_P}" EQUAL "8")
-        set(${out_var} "lib64")
+        set(${out_var} "lib")
       endif()
     endif()
   endif()
diff -Naur a/Modules/Platform/UnixPaths.cmake b/Modules/Platform/UnixPaths.cmake
--- a/Modules/Platform/UnixPaths.cmake	2025-09-30 16:48:33.000000000 +0200
+++ b/Modules/Platform/UnixPaths.cmake	2025-11-05 14:05:48.117279244 +0100
@@ -68,7 +68,7 @@
   )
 
 list(APPEND CMAKE_PLATFORM_IMPLICIT_LINK_DIRECTORIES
-  /lib /lib32 /lib64 /usr/lib /usr/lib32 /usr/lib64
+  /usr/lib
   )
 
 if(CMAKE_SYSROOT_COMPILE)
@@ -106,6 +106,6 @@
 endif()
 
 # Enable use of lib32 and lib64 search path variants by default.
-set_property(GLOBAL PROPERTY FIND_LIBRARY_USE_LIB32_PATHS TRUE)
-set_property(GLOBAL PROPERTY FIND_LIBRARY_USE_LIB64_PATHS TRUE)
-set_property(GLOBAL PROPERTY FIND_LIBRARY_USE_LIBX32_PATHS TRUE)
+set_property(GLOBAL PROPERTY FIND_LIBRARY_USE_LIB32_PATHS FALSE)
+set_property(GLOBAL PROPERTY FIND_LIBRARY_USE_LIB64_PATHS FALSE)
+set_property(GLOBAL PROPERTY FIND_LIBRARY_USE_LIBX32_PATHS FALSE)
diff -Naur a/Source/cmExportInstallCMakeConfigGenerator.cxx b/Source/cmExportInstallCMakeConfigGenerator.cxx
--- a/Source/cmExportInstallCMakeConfigGenerator.cxx	2025-09-30 16:48:33.000000000 +0200
+++ b/Source/cmExportInstallCMakeConfigGenerator.cxx	2025-11-05 14:05:48.117279244 +0100
@@ -161,12 +161,7 @@
     os << "# Compute the installation prefix relative to this file.\n"
        << "get_filename_component(_IMPORT_PREFIX"
        << " \"${CMAKE_CURRENT_LIST_FILE}\" PATH)\n";
-    if (cmHasLiteralPrefix(absDestS, "/lib/") ||
-        cmHasLiteralPrefix(absDestS, "/lib64/") ||
-        cmHasLiteralPrefix(absDestS, "/libx32/") ||
-        cmHasLiteralPrefix(absDestS, "/usr/lib/") ||
-        cmHasLiteralPrefix(absDestS, "/usr/lib64/") ||
-        cmHasLiteralPrefix(absDestS, "/usr/libx32/")) {
+    if (cmHasLiteralPrefix(absDestS, "/usr/lib/")) {
       // Handle "/usr move" symlinks created by some Linux distros.
       /* clang-format off */
       os <<
