diff -Naur a/texk/kpathsea/texmf.cnf b/texk/kpathsea/texmf.cnf
--- a/texk/kpathsea/texmf.cnf	2026-02-07 01:25:39.000000000 +0100
+++ b/texk/kpathsea/texmf.cnf	2026-07-07 02:14:52.322861928 +0200
@@ -59,7 +59,7 @@
 % SELFAUTOPARENT (its grandparent = /usr/local/texlive/YYYY), and
 % SELFAUTOGRANDPARENT (its great-grandparent = /usr/local/texlive).
 % Sorry for the off-by-one-generation names.
-TEXMFROOT = $SELFAUTOPARENT
+TEXMFROOT = /usr/share
 
 % The various texmf trees used by TeX Live, follow.
 % They must must all have the TDS directory structure (https://tug.org/tds).
