diff -Naur a/intl/lwbrk/LineBreaker.cpp b/intl/lwbrk/LineBreaker.cpp
--- a/intl/lwbrk/LineBreaker.cpp	2025-11-06 23:07:23.000000000 +0100
+++ b/intl/lwbrk/LineBreaker.cpp	2025-11-14 20:55:33.665565468 +0100
@@ -453,6 +453,7 @@
       /* AKSARA_START = 45,                 [AS] */ CLASS_CHARACTER,
       /* VIRAMA_FINAL = 46,                 [VF] */ CLASS_CHARACTER,
       /* VIRAMA = 47,                       [VI] */ CLASS_CHARACTER,
+      /* U_LB_UNAMBIGUOUS_HYPHEN = 48,      [HH] */ CLASS_CHARACTER,
   };
 
   static_assert(U_LB_COUNT == std::size(sUnicodeLineBreakToClass),
