diff -Naur a/comm/build/moz.configure/gecko_source.configure b/comm/build/moz.configure/gecko_source.configure
--- a/comm/build/moz.configure/gecko_source.configure	2026-05-15 16:13:17.000000000 +0200
+++ b/comm/build/moz.configure/gecko_source.configure	2026-05-20 02:57:42.873106159 +0200
@@ -51,7 +51,7 @@
                 pass
 
             if len(lines) != 3:
-                log.warn("sourcestamp.txt is corrupt!")
+                log.warning("sourcestamp.txt is corrupt!")
                 return
 
             if lines and lines[line2read].startswith("http"):
