diff -Naur a/justfile b/justfile
--- a/justfile	2025-12-15 12:32:27.000000000 +0100
+++ b/justfile	2025-12-15 20:00:22.025855996 +0100
@@ -10,7 +10,6 @@
     {{ make }} -C cosmic-comp all
     {{ just }} cosmic-edit/build-release
     {{ just }} cosmic-files/build-release
-    {{ just }} cosmic-greeter/build-release
     {{ just }} cosmic-idle/build-release
     {{ just }} cosmic-initial-setup/build-release
     {{ just }} cosmic-launcher/build-release
@@ -23,21 +22,19 @@
     {{ just }} cosmic-settings/build-release
     {{ make }} -C cosmic-settings-daemon all
     {{ just }} cosmic-session/all
-    {{ just }} cosmic-store/build-release
     {{ just }} cosmic-term/build-release
     {{ make }} -C cosmic-wallpapers all
     {{ make }} -C cosmic-workspaces-epoch all
     {{ just }} pop-launcher/build-release
     {{ make }} -C xdg-desktop-portal-cosmic all
 
-install rootdir="" prefix="/usr/local": build
+install rootdir="/var/uhubuild/work/install" prefix="/usr": build
     {{ just }} rootdir={{rootdir}} prefix={{prefix}} cosmic-applets/install
     {{ just }} rootdir={{rootdir}} prefix={{prefix}} cosmic-applibrary/install
     {{ just }} rootdir={{rootdir}} prefix={{prefix}} cosmic-bg/install
     {{ make }} -C cosmic-comp install DESTDIR={{rootdir}} prefix={{prefix}}
     {{ just }} rootdir={{rootdir}} prefix={{prefix}} cosmic-edit/install
     {{ just }} rootdir={{rootdir}} prefix={{prefix}} cosmic-files/install
-    {{ just }} rootdir={{rootdir}} prefix={{prefix}} cosmic-greeter/install
     {{ just }} rootdir={{rootdir}} prefix={{prefix}} cosmic-icons/install
     {{ just }} rootdir={{rootdir}} prefix={{prefix}} cosmic-idle/install
     {{ just }} rootdir={{rootdir}} prefix={{prefix}} cosmic-initial-setup/install
@@ -51,7 +48,6 @@
     {{ just }} rootdir={{rootdir}} prefix={{prefix}} cosmic-settings/install
     {{ make }} -C cosmic-settings-daemon install DESTDIR={{rootdir}} prefix={{prefix}}
     {{ just }} rootdir={{rootdir}} prefix={{prefix}} cosmic-session/install
-    {{ just }} rootdir={{rootdir}} prefix={{prefix}} cosmic-store/install
     {{ just }} rootdir={{rootdir}} prefix={{prefix}} cosmic-term/install
     {{ make }} -C cosmic-wallpapers install DESTDIR={{rootdir}} prefix={{prefix}}
     {{ make }} -C cosmic-workspaces-epoch install DESTDIR={{rootdir}} prefix={{prefix}}
