diff -Naur a/justfile b/justfile
--- a/justfile	2026-02-03 21:07:36.000000000 +0100
+++ b/justfile	2026-02-04 02:44:19.694226506 +0100
@@ -30,7 +30,7 @@
     {{ just }} pop-launcher/build-release
     {{ make }} -C xdg-desktop-portal-cosmic all
 
-install rootdir="" prefix="/usr/local": build
+install rootdir="/var/uhubuild/work/install" prefix="/usr": build
     {{ just }} rootdir={{rootdir}} prefix={{prefix}} cosmic-applets/install
     {{ just }} rootdir={{rootdir}} prefix={{prefix}} cosmic-applibrary/install
     {{ just }} rootdir={{rootdir}} prefix={{prefix}} cosmic-bg/install
