diff -Naur a/justfile b/justfile
--- a/justfile	2026-10-08 00:02:21.000000000 +0200
+++ b/justfile	2026-10-08 01:05:05.232344201 +0200
@@ -7,7 +7,7 @@
     {{ just }} cosmic-applets/build-release
     cd cosmic-app-library && {{ just }} build-release
     {{ just }} cosmic-bg/build-release
-    {{ make }} -C cosmic-comp all
+    {{ just }} cosmic-comp/build-release
     {{ just }} cosmic-edit/build-release
     {{ just }} cosmic-files/build-release
     {{ just }} cosmic-greeter/build-release
@@ -23,7 +23,7 @@
     {{ just }} cosmic-randr/build-release
     {{ just }} cosmic-screenshot/build-release
     cd cosmic-settings && {{ just }} build-release
-    {{ make }} -C cosmic-settings-daemon all
+    cd cosmic-settings-daemon && {{ just }} build-release
     {{ just }} cosmic-session/build-release
     {{ just }} cosmic-store/build-release
     {{ just }} cosmic-term/build-release
@@ -33,11 +33,11 @@
     {{ just }} pop-launcher/build-release
     cd xdg-desktop-portal-cosmic && {{ just }} debug=0
 
-install rootdir="" prefix="/usr/local": build
+install rootdir="/var/uhubuild/work/install" prefix="/usr": build
     {{ just }} rootdir={{rootdir}} prefix={{prefix}} cosmic-applets/install
     {{ just }} rootdir={{rootdir}} prefix={{prefix}} cosmic-app-library/install
     {{ just }} rootdir={{rootdir}} prefix={{prefix}} cosmic-bg/install
-    {{ make }} -C cosmic-comp install DESTDIR={{rootdir}} prefix={{prefix}}
+    {{ just }} rootdir={{rootdir}} prefix={{prefix}} cosmic-comp/install
     {{ just }} rootdir={{rootdir}} prefix={{prefix}} cosmic-edit/install
     {{ just }} rootdir={{rootdir}} prefix={{prefix}} cosmic-files/install
     {{ just }} rootdir={{rootdir}} prefix={{prefix}} cosmic-greeter/install
@@ -54,7 +54,7 @@
     {{ just }} rootdir={{rootdir}} prefix={{prefix}} cosmic-randr/install
     {{ just }} rootdir={{rootdir}} prefix={{prefix}} cosmic-screenshot/install
     {{ just }} rootdir={{rootdir}} prefix={{prefix}} cosmic-settings/install
-    {{ make }} -C cosmic-settings-daemon install DESTDIR={{rootdir}} prefix={{prefix}}
+    {{ just }} rootdir={{rootdir}} prefix={{prefix}} cosmic-settings-daemon/install
     {{ just }} rootdir={{rootdir}} prefix={{prefix}} cosmic-session/install
     {{ just }} rootdir={{rootdir}} prefix={{prefix}} cosmic-store/install
     {{ just }} rootdir={{rootdir}} prefix={{prefix}} cosmic-term/install
