diff -Naur a/conf/main.cf b/conf/main.cf
--- a/conf/main.cf	2025-10-23 21:51:44.000000000 +0200
+++ b/conf/main.cf	2025-12-21 23:57:35.205031005 +0100
@@ -436,8 +436,8 @@
 #
 # The home_mailbox parameter specifies the optional pathname of a
 # mailbox file relative to a user's home directory. The default
-# mailbox file is /var/spool/mail/user or /var/mail/user.  Specify
-# "Maildir/" for qmail-style delivery (the / is required).
+# mailbox file is /var/mail/user.  Specify "Maildir/" for qmail-style
+# delivery (the / is required).
 #
 #home_mailbox = Mailbox
 #home_mailbox = Maildir/
@@ -447,7 +447,6 @@
 # system type.
 #
 #mail_spool_directory = /var/mail
-#mail_spool_directory = /var/spool/mail
 
 # The mailbox_command parameter specifies the optional external
 # command to use instead of mailbox delivery. The command is run as
diff -Naur a/html/local.8.html b/html/local.8.html
--- a/html/local.8.html	2025-01-09 23:18:05.000000000 +0100
+++ b/html/local.8.html	2025-12-21 23:57:35.205031005 +0100
@@ -599,7 +599,7 @@
        The following are examples; details differ between systems.
        $HOME/.forward, per-user aliasing
        /etc/aliases, system-wide alias database
-       /var/spool/mail, system mailboxes
+       /var/mail, system mailboxes
 
 <b><a name="see_also">SEE ALSO</a></b>
        <a href="qmgr.8.html">qmgr(8)</a>, queue manager
diff -Naur a/man/man8/local.8 b/man/man8/local.8
--- a/man/man8/local.8	2025-01-09 23:18:05.000000000 +0100
+++ b/man/man8/local.8	2025-12-21 23:57:35.205031005 +0100
@@ -116,7 +116,7 @@
 .ad
 .fi
 The default per\-user mailbox is a file in the UNIX mail spool
-directory (\fB/var/mail/\fIuser\fR or \fB/var/spool/mail/\fIuser\fR);
+directory (\fB/var/mail/\fIuser\fR);
 the location can be specified with the \fBmail_spool_directory\fR
 configuration parameter. Specify a name ending in \fB/\fR for
 \fBqmail\fR\-compatible \fBmaildir\fR delivery.
@@ -632,7 +632,7 @@
 The following are examples; details differ between systems.
 $HOME/.forward, per\-user aliasing
 /etc/aliases, system\-wide alias database
-/var/spool/mail, system mailboxes
+/var/mail, system mailboxes
 .SH "SEE ALSO"
 .na
 .nf
diff -Naur a/src/local/local.c b/src/local/local.c
--- a/src/local/local.c	2025-01-09 23:18:05.000000000 +0100
+++ b/src/local/local.c	2025-12-21 23:57:35.213031039 +0100
@@ -102,7 +102,7 @@
 /* .ad
 /* .fi
 /*	The default per-user mailbox is a file in the UNIX mail spool
-/*	directory (\fB/var/mail/\fIuser\fR or \fB/var/spool/mail/\fIuser\fR);
+/*	directory (\fB/var/mail/\fIuser\fR);
 /*	the location can be specified with the \fBmail_spool_directory\fR
 /*	configuration parameter. Specify a name ending in \fB/\fR for
 /*	\fBqmail\fR-compatible \fBmaildir\fR delivery.
@@ -586,7 +586,7 @@
 /*	The following are examples; details differ between systems.
 /*	$HOME/.forward, per-user aliasing
 /*	/etc/aliases, system-wide alias database
-/*	/var/spool/mail, system mailboxes
+/*	/var/mail, system mailboxes
 /* SEE ALSO
 /*	qmgr(8), queue manager
 /*	bounce(8), delivery status reports
