diff -Naur a/conf/master.cf b/conf/master.cf
--- a/conf/master.cf	2025-02-05 23:04:14.000000000 +0100
+++ b/conf/master.cf	2025-12-22 00:04:18.842775020 +0100
@@ -116,14 +116,14 @@
 # Also specify in main.cf: cyrus_destination_recipient_limit=1
 #
 #cyrus     unix  -       n       n       -       -       pipe
-#  flags=DRX user=cyrus argv=/cyrus/bin/deliver -e -r ${sender} -m ${extension} ${user}
+#  flags=DRX user=cyrus argv=/usr/libexec/cyrus-imapd/deliver -e -r ${sender} -m ${extension} ${user}
 #
 # ====================================================================
 #
 # Old example of delivery via Cyrus.
 #
 #old-cyrus unix  -       n       n       -       -       pipe
-#  flags=R user=cyrus argv=/cyrus/bin/deliver -e -m ${extension} ${user}
+#  flags=R user=cyrus rgv=/usr/libexec/cyrus-imapd/deliver -e -m ${extension} ${user}
 #
 # ====================================================================
 #
