diff -Naur a/conf/main.cf b/conf/main.cf
--- a/conf/main.cf	2025-12-22 00:02:36.882335521 +0100
+++ b/conf/main.cf	2025-12-22 00:05:39.219121003 +0100
@@ -487,7 +487,7 @@
 #
 # Cyrus IMAP over LMTP. Specify ``lmtpunix      cmd="lmtpd"
 # listen="/var/imap/socket/lmtp" prefork=0'' in cyrus.conf.
-#mailbox_transport = lmtp:unix:/var/imap/socket/lmtp
+#mailbox_transport = lmtp:unix:/var/lib/imap/socket/lmtp
 #
 # Cyrus IMAP via command line. Uncomment the "cyrus...pipe" and
 # subsequent line in master.cf.
