diff -Naur a/conf/main.cf b/conf/main.cf
--- a/conf/main.cf	2025-12-22 00:05:39.219121003 +0100
+++ b/conf/main.cf	2025-12-22 00:07:48.499676630 +0100
@@ -681,6 +681,18 @@
 # readme_directory: The location of the Postfix README files.
 #
 readme_directory =
+
+# Greylisting
+#
+# restart the policy server less frequently so
+#
+#policy_time_limit = 3600
+
+#smtpd_recipient_restrictions =
+#  reject_unlisted_recipient
+#  reject_unauth_destination
+#  check_policy_service unix:private/policy
+
 inet_protocols = ipv4
 shlib_directory = /usr/lib/postfix/${mail_version}
 meta_directory = /etc/postfix
diff -Naur a/conf/master.cf b/conf/master.cf
--- a/conf/master.cf	2025-12-22 00:04:18.842775020 +0100
+++ b/conf/master.cf	2025-12-22 00:07:48.503676647 +0100
@@ -149,3 +149,9 @@
 #mailman   unix  -       n       n       -       -       pipe
 #  flags=FRX user=list argv=/usr/lib/mailman/bin/postfix-to-mailman.py
 #  ${nexthop} ${user}
+#
+# Policy server (greylist.pl)
+#
+#policy  unix  -       n       n       -       -       spawn
+#  user=nobody argv=/usr/bin/perl /usr/lib/postfix/greylist.pl
+#
diff -Naur a/examples/smtpd-policy/greylist.pl b/examples/smtpd-policy/greylist.pl
--- a/examples/smtpd-policy/greylist.pl	2007-01-04 23:30:26.000000000 +0100
+++ b/examples/smtpd-policy/greylist.pl	2025-12-22 00:07:48.503676647 +0100
@@ -73,7 +73,7 @@
 # In case of database corruption, this script saves the database as
 # $database_name.time(), so that the mail system does not get stuck.
 #
-$database_name="/var/mta/greylist.db";
+$database_name="/var/lib/greylist/greylist.db"
 $greylist_delay=60;
 
 #
