diff -Naur a/conf/master.cf b/conf/master.cf
--- a/conf/master.cf	2025-12-22 00:07:48.503676647 +0100
+++ b/conf/master.cf	2025-12-22 00:11:10.148540310 +0100
@@ -155,3 +155,5 @@
 #policy  unix  -       n       n       -       -       spawn
 #  user=nobody argv=/usr/bin/perl /usr/lib/postfix/greylist.pl
 #
+dovecot   unix  -       n       n       -       -       pipe
+  flags=DRhu user=vmail:nogroup argv=/usr/libexec/dovecot/deliver -f ${sender} -d ${recipient}
