diff -Naur a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	2026-06-11 11:31:53.000000000 +0200
+++ b/CMakeLists.txt	2026-06-20 19:22:36.071566781 +0200
@@ -111,8 +111,6 @@
     TYPE RUNTIME
 )
 
-find_package(Corrosion REQUIRED)
-
 add_definitions(
     -DQT_DEPRECATED_WARNINGS
     -DQT_NO_URL_CAST_FROM_STRING
diff -Naur a/kdeds/kameleon/qmk/kameleon-qmk-helper/CMakeLists.txt b/kdeds/kameleon/qmk/kameleon-qmk-helper/CMakeLists.txt
--- a/kdeds/kameleon/qmk/kameleon-qmk-helper/CMakeLists.txt	2026-06-11 11:31:53.000000000 +0200
+++ b/kdeds/kameleon/qmk/kameleon-qmk-helper/CMakeLists.txt	2026-06-20 19:22:36.071775524 +0200
@@ -1,6 +1,14 @@
 # SPDX-License-Identifier: BSD-2-Clause
 # SPDX-FileCopyrightText: 2026 Harald Sitter <sitter@kde.org>
 
+include(FetchContent)
+
+FetchContent_Declare(
+    Corrosion
+    GIT_REPOSITORY https://github.com/corrosion-rs/corrosion.git
+)
+FetchContent_MakeAvailable(Corrosion)
+
 # Import targets defined in a package or workspace manifest `Cargo.toml` file
 corrosion_import_crate(MANIFEST_PATH Cargo.toml)
 
