diff -Naur a/lib/dpkg/dpkg.h b/lib/dpkg/dpkg.h
--- a/lib/dpkg/dpkg.h	2026-03-01 13:52:09.368783951 +0100
+++ b/lib/dpkg/dpkg.h	2026-03-01 13:56:35.061803630 +0100
@@ -71,6 +71,7 @@
 #define POSTINSTFILE       "postinst"
 #define PRERMFILE          "prerm"
 #define POSTRMFILE         "postrm"
+#define POSTRMCUSTOMFILE   "postrm-custom"
 /* Debconf config maintainer script. */
 #define MAINTSCRIPT_FILE_CONFIG		"config"
 #define TRIGGERSCIFILE     "triggers"
diff -Naur a/src/main/remove.c b/src/main/remove.c
--- a/src/main/remove.c	2026-01-23 01:47:06.000000000 +0100
+++ b/src/main/remove.c	2026-03-01 14:01:01.514822734 +0100
@@ -243,7 +243,8 @@
 {
 	/* We need the postrm and list files for --purge. */
 	if (strcmp(filetype, LISTFILE) == 0 ||
-	    strcmp(filetype, POSTRMFILE) == 0)
+	    strcmp(filetype, POSTRMFILE) == 0 ||
+	    strcmp(filetype, POSTRMCUSTOMFILE) == 0)
 		return;
 
 	debug_at(dbg_stupidlyverbose, "metadata file not postrm nor list");
@@ -732,6 +733,12 @@
 		if (unlink(filename) && errno != ENOENT)
 			ohshite(_("cannot remove old postrm script"));
 
+		filename = pkg_infodb_get_file(pkg, &pkg->installed, POSTRMCUSTOMFILE);
+		debug_at(dbg_general, "purge done, removing postrm-custom '%s'",
+		         filename);
+		if (unlink(filename) && errno != ENOENT)
+			ohshite(_("cannot remove old postrm-custom script"));
+
 		pkg_set_status(pkg, PKG_STAT_NOTINSTALLED);
 		pkg_set_want(pkg, PKG_WANT_UNKNOWN);
 
