diff -Naur a/src/helpers/ub_compile b/src/helpers/ub_compile
--- a/src/helpers/ub_compile	2025-08-18 23:58:56.000000000 +0200
+++ b/src/helpers/ub_compile	2025-11-06 03:01:19.199534126 +0100
@@ -1,9 +1,5 @@
 #!/bin/bash -eu
 
-# a ccache bezavarhat a configure lépésnél
-# pl. webkit2gtk nem találja a pthread.h-t
-export DISABLE_CCACHE=1
-
 if [ -f meson.build ]; then
 	argv=()
 	argx=(
diff -Naur a/src/helpers/ub_make b/src/helpers/ub_make
--- a/src/helpers/ub_make	2025-08-18 23:58:56.000000000 +0200
+++ b/src/helpers/ub_make	2025-11-06 03:02:18.255752916 +0100
@@ -4,8 +4,6 @@
 
 j="-j${UB_MAKE_PROCS:-$(_procs)}"
 
-export DISABLE_CCACHE=0
-
 if [ -f meson.build -a -f build/build.ninja ]; then
 	echo "ninja $j"
 	for i in "$@"; do
