diff -Naur a/bash-completion/completions/build b/bash-completion/completions/build
--- a/bash-completion/completions/build	2025-08-18 23:58:56.000000000 +0200
+++ b/bash-completion/completions/build	1970-01-01 01:00:00.000000000 +0100
@@ -1,26 +0,0 @@
-_buildubk3()
-{
-        local cur foo
-        cur=${COMP_WORDS[COMP_CWORD]}
-        foo="$(/bin/ls /usr/src/UHUBUILD/UB-UBK3)"
-        COMPREPLY=($(compgen -W "$foo" -- "$cur"))
-}
-complete -F _buildubk3 build-ubk3
-
-_buildubk4()
-{
-        local cur foo
-        cur=${COMP_WORDS[COMP_CWORD]}
-        foo="$(/bin/ls /usr/src/UHUBUILD/UB-UBK4)"
-        COMPREPLY=($(compgen -W "$foo" -- "$cur"))
-}
-complete -F _buildubk4 build-ubk4
-
-_buildubk5()
-{
-        local cur foo
-        cur=${COMP_WORDS[COMP_CWORD]}
-        foo="$(/bin/ls /usr/src/UHUBUILD/UB-UBK5)"
-        COMPREPLY=($(compgen -W "$foo" -- "$cur"))
-}
-complete -F _buildubk5 build-ubk5
diff -Naur a/src/build/build b/src/build/build
--- a/src/build/build	2025-11-06 12:13:06.410059381 +0100
+++ b/src/build/build	2025-11-06 22:57:58.261048961 +0100
@@ -8,7 +8,7 @@
 # Az első verziót összeállította: Sári Gábor <saga@kde.hu>
 # 2004. január 19. táján a II. UHU Csomagoló Party után
 #
-# A legfrissebb verzió elérhető a http://ubk.hu/ oldalról.
+# A legfrissebb verzió elérhető a http://rezso.net/ oldalról.
 #
 # A Changelog a csomag forrásában található
 #
@@ -25,14 +25,12 @@
 
 case `basename $0` in
 	build)
-		export UHUBUILD_DISTR=UBK5;;
-	build-ubk4)
-		export UHUBUILD_DISTR=UBK4;;
-	build-ubk5)
-		export UHUBUILD_DISTR=UBK5;;
+		export UHUBUILD_DISTR=RN1;;
+	build-rn1)
+		export UHUBUILD_DISTR=RN1;;
 esac
 
-UHUBUILD_DISTR_NUM=${UHUBUILD_DISTR/UBK/}
+UHUBUILD_DISTR_NUM=${UHUBUILD_DISTR/RN/}
 
 dieifuser
 
@@ -203,7 +201,7 @@
 			appleecho "Az első verziót összeállította: Sári Gábor <saga@kde.hu>"
 			appleecho "2004. január 19. táján a II. UHU Csomagoló Party után"
 			echo
-			appleecho "A legfrissebb verzió elérhető a http://ubk.hu/ oldalról."
+			appleecho "A legfrissebb verzió elérhető a http://rezso.net/ oldalról."
 			exit 0;;
 		esac
 done
diff -Naur a/src/build/build-all b/src/build/build-all
--- a/src/build/build-all	2025-11-06 12:13:06.410059381 +0100
+++ b/src/build/build-all	1970-01-01 01:00:00.000000000 +0100
@@ -1,13 +0,0 @@
-#!/bin/bash
-
-# UHUBUILD környezetben verziófüggetlen
-# fordítást levezénylő szkript
-#
-# Összeállította: Sári Gábor <saga@kde.hu>
-# 2004. január 19. táján a II. UHU Csomagoló Party után
-
-dieifuser
-
-for i in ubk4 ubk5; do
-	/usr/bin/build-"$i" "$@"
-done
diff -Naur a/src/build/build-clean b/src/build/build-clean
--- a/src/build/build-clean	2025-11-06 12:13:06.410059381 +0100
+++ b/src/build/build-clean	2025-11-06 22:58:08.293075981 +0100
@@ -17,11 +17,9 @@
 
 case ${BASENAME} in
 	build-clean)
-		export DISTR=UBK5;;
-	build-clean-ubk4)
-		export DISTR=UBK4;;
-	build-clean-ubk5)
-		export DISTR=UBK5;;
+		export DISTR=RN1;;
+	build-clean-rn1)
+		export DISTR=RN1;;
 esac
 
 dieifuser
diff -Naur a/src/build/build-clean-all b/src/build/build-clean-all
--- a/src/build/build-clean-all	2025-11-06 12:13:06.410059381 +0100
+++ b/src/build/build-clean-all	1970-01-01 01:00:00.000000000 +0100
@@ -1,11 +0,0 @@
-#!/bin/bash
-
-# UHUBUILD környezet verziófüggetlen takarító szkript
-#
-# A szkriptet eredetileg készítette: Charlos, 2007. március 9.-én.
-
-dieifuser
-
-for i in ubk4 ubk5; do
-	/usr/bin/build-clean-"$i" "$@"
-done
diff -Naur a/src/uhubuild-build b/src/uhubuild-build
--- a/src/uhubuild-build	2025-11-06 12:13:06.402059369 +0100
+++ b/src/uhubuild-build	2025-11-06 22:49:00.368312428 +0100
@@ -69,7 +69,7 @@
 export UB_PACKAGES=$(cd "$UB_SRCDIR/packages"; echo *)
 export UB_SOURCENAME="$(<"$UB_SRCDIR/sourcename")"
 export UB_DISTRIBUTION="$(<"$UB_SRCDIR/distribution")"
-export UB_DISTRIBUTION_NUM=${UB_DISTRIBUTION/UHU-Linux UBK/}
+export UB_DISTRIBUTION_NUM=${UB_DISTRIBUTION/UHU-Linux RN/}
 export UB_FULLVERSION="$(<"$UB_SRCDIR/version")"
 if grep -q ":" "$UB_SRCDIR/version"; then
 export UB_VERSION=${UB_FULLVERSION/*:/}
