From ab9c30fe0d45ae3aeb99c674d2415a5fe28d388c Mon Sep 17 00:00:00 2001
From: rezso <rezso@rezso.net>
Date: Fri, 7 Nov 2025 01:28:59 +0100
Subject: nem használt változó törlése
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit


diff --git a/src/build/build b/src/build/build
index a2d5646..f7f2e5d 100755
--- a/src/build/build
+++ b/src/build/build
@@ -30,8 +30,6 @@ case `basename $0` in
 		export UHUBUILD_DISTR=RN1;;
 esac
 
-UHUBUILD_DISTR_NUM=${UHUBUILD_DISTR/RN/}
-
 dieifuser
 
 export UB_ARCH="amd64"
diff --git a/src/uhubuild-build b/src/uhubuild-build
index db11425..e7a90bf 100755
--- a/src/uhubuild-build
+++ b/src/uhubuild-build
@@ -69,7 +69,6 @@ mkdir -p "$PRETENDROOTDIR" "$UB_TMPDIR"
 export UB_PACKAGES=$(cd "$UB_SRCDIR/packages"; echo *)
 export UB_SOURCENAME="$(<"$UB_SRCDIR/sourcename")"
 export UB_DISTRIBUTION="$(<"$UB_SRCDIR/distribution")"
-export UB_DISTRIBUTION_NUM=${UB_DISTRIBUTION/UHU-Linux RN/}
 export UB_FULLVERSION="$(<"$UB_SRCDIR/version")"
 if grep -q ":" "$UB_SRCDIR/version"; then
 export UB_VERSION=${UB_FULLVERSION/*:/}
