diff -Naur a/src/steps/depends b/src/steps/depends
--- a/src/steps/depends	2026-03-30 14:10:48.000000000 +0200
+++ b/src/steps/depends	2026-04-16 22:37:31.560639262 +0200
@@ -355,21 +355,29 @@
 			    found=0
 				for cmakedep in `grep ^find_dependency $file | sed 's/.*find_dependency[(]//g' | sed 's/[)]//g' | sed 's/\s.*//g'`; do
 				    # echo ${cmakedep}
-					cmakefile="${cmakedep}/${cmakedep}Config.cmake"               
-					if [ -f "/usr/lib/cmake/${cmakefile}" ]; then
-						echo "/usr/lib/cmake/${cmakefile}" >> "$UB_TMPDIR/file-noversion"
+					cmakefile="${cmakedep}Config.cmake"               
+					if [ -f "/usr/lib/cmake/${cmakedep}/${cmakefile}" ]; then
+						echo "/usr/lib/cmake/${cmakedep}/${cmakefile}" >> "$UB_TMPDIR/file-noversion"
 					    found=1
 					fi
-					if [ -f "/usr/share/cmake/${cmakefile}" ]; then
-						echo "/usr/share/cmake/${cmakefile}" >> "$UB_TMPDIR/file-noversion"
+					if [ -f "/usr/share/cmake/${cmakedep}/${cmakefile}" ]; then
+						echo "/usr/share/cmake/${cmakedep}/${cmakefile}" >> "$UB_TMPDIR/file-noversion"
 					    found=1
 					fi
-					if [ -f "/usr/share/ECM/${cmakefile}" ]; then
-						echo "/usr/share/ECM/${cmakefile}" >> "$UB_TMPDIR/file-noversion"
+					if [ -f "/usr/lib/cmake/${cmakedep,,}/${cmakefile}" ]; then
+						echo "/usr/lib/cmake/${cmakedep,,}/${cmakefile}" >> "$UB_TMPDIR/file-noversion"
 					    found=1
 					fi
-					if [ -f "/usr/share/${cmakefile}" ]; then
-						echo "/usr/share/${cmakefile}" >> "$UB_TMPDIR/file-noversion"
+					if [ -f "/usr/share/cmake/${cmakedep,,}/${cmakefile}" ]; then
+						echo "/usr/share/cmake/${cmakedep,,}/${cmakefile}" >> "$UB_TMPDIR/file-noversion"
+					    found=1
+					fi
+					if [ -f "/usr/share/ECM/${cmakedep}/${cmakefile}" ]; then
+						echo "/usr/share/ECM/${cmakedep}/${cmakefile}" >> "$UB_TMPDIR/file-noversion"
+					    found=1
+					fi
+					if [ -f "/usr/share/${cmakedep}/${cmakefile}" ]; then
+						echo "/usr/share/${cmakedep}/${cmakefile}" >> "$UB_TMPDIR/file-noversion"
 					    found=1
 					fi
     				if [ -f "/usr/lib/pkgconfig/${cmakedep}.pc" ]; then
@@ -394,16 +402,20 @@
 					    found=1
 					fi
 					for p in $UB_PACKAGES; do
-					    if [ -f "$UB_PACKAGEDIR/$p/usr/lib/cmake/${cmakefile}" ]; then
-						    echo "/usr/lib/cmake/${cmakefile}" >> "$UB_TMPDIR/file-noversion"
+					    if [ -f "$UB_PACKAGEDIR/$p/usr/lib/cmake/${cmakedep}/${cmakefile}" ]; then
+						    echo "/usr/lib/cmake/${cmakedep}/${cmakefile}" >> "$UB_TMPDIR/file-noversion"
+						    found=1
+						fi
+					    if [ -f "$UB_PACKAGEDIR/$p/usr/share/cmake/${cmakedep}/${cmakefile}" ]; then
+						    echo "/usr/share/cmake/${cmakedep}/${cmakefile}" >> "$UB_TMPDIR/file-noversion"
 						    found=1
 						fi
-					    if [ -f "$UB_PACKAGEDIR/$p/usr/share/cmake/${cmakefile}" ]; then
-						    echo "/usr/share/cmake/${cmakefile}" >> "$UB_TMPDIR/file-noversion"
+					    if [ -f "$UB_PACKAGEDIR/$p/usr/lib/cmake/${cmakedep,,}/${cmakefile}" ]; then
+						    echo "/usr/lib/cmake/${cmakedep,,}/${cmakefile}" >> "$UB_TMPDIR/file-noversion"
 						    found=1
 						fi
-					    if [ -f "$UB_PACKAGEDIR/$p//usr/share/${cmakefile}" ]; then
-						    echo "/usr/share/${cmakefile}" >> "$UB_TMPDIR/file-noversion"
+					    if [ -f "$UB_PACKAGEDIR/$p//usr/share/${cmakedep,,}/${cmakefile}" ]; then
+						    echo "/usr/share/${cmakedep}/${cmakefile}" >> "$UB_TMPDIR/file-noversion"
 						    found=1
 						fi
 					    if [ -f "$UB_PACKAGEDIR/$p/usr/share/ECM/cmake/${cmakedep}Config.cmake" ]; then
