diff -Naur a/debian/synaptic-pkexec b/debian/synaptic-pkexec
--- a/debian/synaptic-pkexec	2026-10-05 20:32:41.000000000 +0200
+++ b/debian/synaptic-pkexec	2026-10-07 08:07:31.943458128 +0200
@@ -1,7 +1,13 @@
 #!/bin/sh
 
 if command -v pkexec >/dev/null 2>&1; then
-	exec pkexec "/usr/bin/synaptic" "$@"
+    if [ x"${XDG_SESSION_TYPE}" = "xx11" ]; then
+    	exec pkexec "/usr/bin/synaptic" "$@"
+    else
+        xhost +si:localuser:root
+        pkexec "/usr/bin/synaptic" "$@"
+        xhost -si:localuser:root
+    fi
 fi
 
 if command -v run0 >/dev/null 2>&1; then
