diff -Naur a/src/modules/bootloader/bootloader.conf b/src/modules/bootloader/bootloader.conf
--- a/src/modules/bootloader/bootloader.conf	2025-07-21 18:32:10.000000000 +0200
+++ b/src/modules/bootloader/bootloader.conf	2025-12-13 18:03:47.095604635 +0100
@@ -21,7 +21,7 @@
 # kernelSearchPath is the path relative to the root of the install to search for kernels
 # A kernel is identified by finding files which match regular expression, kernelPattern
 kernelSearchPath: "/usr/lib/modules"
-kernelPattern: "^vmlinuz.*"
+kernelPattern: "^kernel.*"
 
 # loaderEntries is an array of options to add to loader.conf for systemd-boot
 # please note that the "default" option is added programmatically
