diff -Naur a/Makefile b/Makefile
--- a/Makefile	2025-06-21 18:06:20.000000000 +0200
+++ b/Makefile	2026-02-26 01:52:17.057577167 +0100
@@ -19,7 +19,7 @@
 DNS=
 
 # Build libpci as a shared library (yes/no; or local for testing; requires GCC)
-SHARED=no
+SHARED=yes
 
 # Use libkmod to resolve kernel modules on Linux (yes/no, default: detect)
 LIBKMOD=
@@ -32,7 +32,7 @@
 ABI_VERSION=3
 
 # Installation directories
-PREFIX=/usr/local
+PREFIX=/usr
 BINDIR=$(PREFIX)/bin
 SBINDIR=$(PREFIX)/sbin
 SHAREDIR=$(PREFIX)/share
@@ -159,7 +159,7 @@
 
 distclean: clean
 
-install: all
+install: all install-lib
 # -c is ignored on Linux, but required on FreeBSD
 	$(DIRINSTALL) -m 755 $(DESTDIR)$(BINDIR) $(DESTDIR)$(SBINDIR) $(DESTDIR)$(IDSDIR) $(DESTDIR)$(MANDIR)/man8 $(DESTDIR)$(MANDIR)/man7 $(DESTDIR)$(MANDIR)/man5
 	$(INSTALL) -c -m 755 $(STRIP) lspci$(EXEEXT) $(DESTDIR)$(LSPCIDIR)
