diff -Naur a/browser/app/profile/firefox.js b/browser/app/profile/firefox.js
--- a/browser/app/profile/firefox.js	2026-08-12 23:13:24.000000000 +0200
+++ b/browser/app/profile/firefox.js	2026-08-18 23:14:19.024351291 +0200
@@ -18,6 +18,12 @@
   #endif
 #endif
 
+// Use LANG environment variable to choose locale
+pref("intl.locale.requested", "");
+
+// Use system-provided dictionaries
+pref("spellchecker.dictionary_path", "/usr/share/hunspell");
+
 #ifdef XP_MACOSX
   pref("browser.hiddenWindowChromeURL", "chrome://browser/content/hiddenWindowMac.xhtml");
 #endif
@@ -57,7 +63,8 @@
 
 // Disable add-ons that are not installed by the user in all scopes by default.
 // See the SCOPE constants in AddonManager.sys.mjs for values to use here.
-pref("extensions.autoDisableScopes", 15);
+pref("extensions.autoDisableScopes", 11);
+pref("extensions.showMismatchUI", false);
 // Scopes to scan for changes at startup.
 pref("extensions.startupScanScopes", 0);
 
@@ -277,7 +284,7 @@
 pref("browser.compactmode.show", false);
 
 // At startup, check if we're the default browser and prompt user if not.
-pref("browser.shell.checkDefaultBrowser", true);
+pref("browser.shell.checkDefaultBrowser", false);
 pref("browser.shell.shortcutFavicons",true);
 pref("browser.shell.mostRecentDateSetAsDefault", "");
 pref("browser.shell.skipDefaultBrowserCheckOnFirstRun", true);
@@ -398,7 +405,7 @@
 pref("browser.urlbar.accessibility.tabToSearch.announceResults", true);
 
 // Control autoFill behavior
-pref("browser.urlbar.autoFill", true);
+pref("browser.urlbar.autoFill", false);
 
 // Whether enabling adaptive history autofill.
 #ifdef NIGHTLY_BUILD
@@ -662,7 +669,7 @@
 // is longer than this.
 pref("browser.urlbar.maxCharsForSearchSuggestions", 100);
 
-pref("browser.urlbar.trimURLs", true);
+pref("browser.urlbar.trimURLs", false);
 pref("browser.urlbar.trimHttps", false);
 pref("browser.urlbar.untrimOnUserInteraction.featureGate", false);
 
@@ -1472,7 +1479,7 @@
 // 2 and other values, nothing
 pref("browser.backspace_action", 2);
 
-pref("intl.regional_prefs.use_os_locales", false);
+pref("intl.regional_prefs.use_os_locales", true);
 
 pref("browser.send_pings", false);
 
@@ -2535,6 +2542,9 @@
 
 pref("browser.contentblocking.database.enabled", true);
 
+// Enable URL query stripping
+pref("privacy.query_stripping.enabled", true);
+
 // Enable Strip on Share by default on desktop
 pref("privacy.query_stripping.strip_on_share.enabled", true);
 
