diff -Naur a/completions-fallback/dpkg.bash b/completions-fallback/dpkg.bash
--- a/completions-fallback/dpkg.bash	2026-07-05 10:06:41.000000000 +0200
+++ b/completions-fallback/dpkg.bash	2026-07-17 03:41:21.548186413 +0200
@@ -80,7 +80,7 @@
     case $prev in
         --install | --unpack | --record-avail | --contents | --info | --fsys-tarfile | \
             --field | --control | --extract | --vextract | --raw-extract | -${noargopts}[ciAIfexXR])
-            _comp_compgen_filedir '?(u|d)deb'
+            _comp_compgen_filedir '?(u|d)deb|uhu'
             return
             ;;
         --build | --admindir | --instdir | --root | -${noargopts}b)
@@ -95,7 +95,7 @@
             if [[ $1 == *dpkg-query ]]; then
                 _comp_compgen -x apt-cache packages
             else
-                _comp_compgen_filedir '?(u|d)deb'
+                _comp_compgen_filedir '?(u|d)deb|uhu'
             fi
             return
             ;;
