diff -Naur a/src/useradd.c b/src/useradd.c
--- a/src/useradd.c	2026-10-05 10:15:53.853433847 +0200
+++ b/src/useradd.c	2026-10-05 10:17:06.680063841 +0200
@@ -179,7 +179,7 @@
     lflg = false,		/* do not add user to lastlog/faillog databases */
     mflg = false,		/* create user's home directory if it doesn't exist */
     Mflg = false,		/* do not create user's home directory even if CREATE_HOME is set */
-    Nflg = false,		/* do not create a group having the same name as the user, but add the user to def_group (or the group specified with -g) */
+    Nflg = true,		/* do not create a group having the same name as the user, but add the user to def_group (or the group specified with -g) */
     oflg = false,		/* permit non-unique user ID to be specified with -u */
     rflg = false,		/* create a system account */
     sflg = false,		/* shell program for new account */
