diff -Naur a/Makefile b/Makefile
--- a/Makefile	2026-04-21 09:20:22.000000000 +0200
+++ b/Makefile	2026-04-21 10:56:44.563822231 +0200
@@ -13,9 +13,10 @@
 
 ADD_CFLAGS  = # -DVI_KEYS
 
-DESTDIR     = /usr
-BINDIR      = $(DESTDIR)/bin
-MANDIR      = $(DESTDIR)/share/man/man1
+DESTDIR     =
+PREFIX      = /usr
+BINDIR      = $(DESTDIR)$(PREFIX)/bin
+MANDIR      = $(DESTDIR)$(PREFIX)/share/man/man1
 
 
 # Uncomment the lines for your system (default is linux)
@@ -26,7 +27,7 @@
 COLOR       = -DCOLOR_SUPPORT 
 READLINE    = -DREADLINE_SUPPORT
 CFLAGS      += -D_GNU_SOURCE $(COLOR) $(READLINE) $(ADD_CFLAGS)
-LDFLAGS     += -lncurses -ltinfo -lreadline
+LDFLAGS     += -lncursesw -lreadline
 
 ###########################
 # LINUX with Wide character support (tested with ncurses 6)
