diff -Naur a/Configurations/unix-Makefile.tmpl b/Configurations/unix-Makefile.tmpl
--- a/Configurations/unix-Makefile.tmpl	2026-01-27 15:09:55.000000000 +0100
+++ b/Configurations/unix-Makefile.tmpl	2026-02-05 19:36:10.084024974 +0100
@@ -513,7 +513,7 @@
 {- dependmagic('build_programs'); -}: build_programs_nodep
 
 build_generated_pods: $(GENERATED_PODS)
-build_docs: build_man_docs build_html_docs
+build_docs: build_man_docs
 build_man_docs: $(MANDOCS1) $(MANDOCS3) $(MANDOCS5) $(MANDOCS7)
 build_html_docs: $(HTMLDOCS1) $(HTMLDOCS3) $(HTMLDOCS5) $(HTMLDOCS7)
 
@@ -623,7 +623,7 @@
 
 uninstall_sw: uninstall_runtime uninstall_modules uninstall_engines uninstall_dev
 
-install_docs: install_man_docs install_html_docs
+install_docs: install_man_docs
 
 uninstall_docs: uninstall_man_docs uninstall_html_docs
 	$(RM) -r "$(DESTDIR)$(DOCDIR)"
