diff -Naur a/build/config/compiler/BUILD.gn b/build/config/compiler/BUILD.gn
--- a/build/config/compiler/BUILD.gn	2026-08-15 01:34:34.200577165 +0200
+++ b/build/config/compiler/BUILD.gn	2026-08-15 01:59:52.046441187 +0200
@@ -578,7 +578,7 @@
     cflags += [ "-fcolor-diagnostics" ]
     if (!is_wasm) {
       if (!is_win) {
-        cflags += [ "-fdiagnostics-show-inlining-chain" ]
+        cflags += [ ]
       } else {
         # Combine after https://github.com/llvm/llvm-project/pull/192241
         cflags += [ "/clang:-fdiagnostics-show-inlining-chain" ]
@@ -614,13 +614,6 @@
       ]
     }
 
-    # The performance improvement does not seem worth the risk. See
-    # https://crbug.com/484082200 for background and https://crrev.com/c/7593035
-    # for discussion.
-    if (!is_wasm) {
-      cflags += [ "-fno-lifetime-dse" ]
-    }
-
     # TODO(hans): Remove this once Clang generates better optimized debug info
     # by default. https://crbug.com/765793
     cflags += [
@@ -1578,7 +1571,7 @@
 # See also: https://crbug.com/40891132#comment10
 ubsan_hardening("c_array_bounds") {
   sanitizer = "array-bounds"
-  condition = !(is_asan && target_cpu == "x86") && !is_wasm
+  condition = false
 
   # Because we've enabled array-bounds sanitizing we also want to suppress
   # the related warning about "unsafe-buffer-usage-in-static-sized-array",
@@ -1592,7 +1585,7 @@
 # `NOTREACHED()` at the end of such functions.
 ubsan_hardening("return") {
   sanitizer = "return"
-  condition = !is_wasm
+  condition = false
 }
 
 config("rustc_revision") {
@@ -1933,9 +1926,6 @@
       cflags += [
         # TODO(crbug.com/432275627): Fix and re-enable.
         "-Wno-uninitialized-const-pointer",
-
-        # TODO(crbug.com/495753203): Fix and re-enable.
-        "-Wno-unused-but-set-global",
       ]
     }
 
