diff -Naur a/backends/apt/meson.build b/backends/apt/meson.build
--- a/backends/apt/meson.build	2026-01-28 20:54:23.000000000 +0100
+++ b/backends/apt/meson.build	2026-02-04 01:59:44.756184650 +0100
@@ -119,35 +119,4 @@
   install_dir: join_paths(get_option('datadir'), 'PackageKit', 'helpers', 'apt'),
 )
 
-if get_option('systemd')
-  executable(
-    'pk-debconf-helper',
-    'pk-debconf-helper.c',
-    dependencies: [
-      packagekit_glib2_dep,
-      libsystemd
-    ],
-    install: true,
-    install_dir: get_option('libexecdir')
-  )
-
-  sd_config_data = configuration_data()
-  sd_config_data.set('libexecdir', join_paths(get_option('prefix'), get_option('libexecdir')))
-
-  configure_file(
-    input: 'pk-debconf-helper.service.in',
-    output: 'pk-debconf-helper.service',
-    configuration: sd_config_data,
-    install: true,
-    install_dir: systemd_user_unit_dir,
-  )
-  configure_file(
-    input: 'pk-debconf-helper.socket.in',
-    output: 'pk-debconf-helper.socket',
-    configuration: sd_config_data,
-    install: true,
-    install_dir: systemd_user_unit_dir,
-  )
-endif
-
 subdir('tests')
