diff -Naur a/etc/mplayer.desktop b/etc/mplayer.desktop
--- a/etc/mplayer.desktop	2026-06-29 23:00:50.000000000 +0200
+++ b/etc/mplayer.desktop	2026-07-13 13:08:35.854586099 +0200
@@ -6,6 +6,7 @@
 GenericName[de]=Medienwiedergabe
 GenericName[es]=Reproductor multimedia
 GenericName[fr]=Lecteur multimédia
+GenericName[hu]=Médialejátszó
 GenericName[it]=Lettore multimediale
 GenericName[ja]=メディアプレーヤー
 X-GNOME-FullName=MPlayer Media Player
