diff -Naur a/build/config/rust.gni b/build/config/rust.gni
--- a/build/config/rust.gni	2026-01-07 01:50:30.000000000 +0100
+++ b/build/config/rust.gni	2026-01-15 15:56:26.334490532 +0100
@@ -121,7 +121,7 @@
 # `rustc_nightly_capability = false` will make the build avoid the use of
 # Rust nightly features. There are no bots that test this and there is no
 # guarantee it will work, but we accept patches for this configuration.
-rustc_nightly_capability = use_chromium_rust_toolchain
+rustc_nightly_capability = true
 
 # Platform support for the Rust toolchain.
 chromium_toolchain_supports_platform = !is_wasm
