diff -Naur a/build/config/compiler/pgo/pgo.gni b/build/config/compiler/pgo/pgo.gni
--- a/build/config/compiler/pgo/pgo.gni	2026-05-01 19:13:59.000000000 +0200
+++ b/build/config/compiler/pgo/pgo.gni	2026-05-10 21:48:17.898845292 +0200
@@ -24,12 +24,6 @@
   # to think some code is hotter than it actually is, potentially causing very
   # bad compile times.
   chrome_pgo_phase = 0
-  if (!is_cronet_build && !dcheck_always_on && is_official_build &&
-      (is_high_end_android || is_win || is_mac || is_fuchsia ||
-       (is_linux && !is_castos) ||
-       (is_ios && use_blink && target_environment == "device"))) {
-    chrome_pgo_phase = 2
-  }
 
   # When using chrome_pgo_phase = 2, read profile data from this path.
   pgo_data_path = ""
