diff -Naur a/methods/gpgv.cc b/methods/gpgv.cc
--- a/methods/gpgv.cc	2025-10-04 14:30:16.000000000 +0200
+++ b/methods/gpgv.cc	2025-10-19 22:24:51.122413396 +0200
@@ -70,13 +70,13 @@
 
 static constexpr std::array<Digest,12> Digests = {{
    {Digest::State::Untrusted, "Invalid digest"},
-   {Digest::State::Untrusted, "MD5"},
-   {Digest::State::Untrusted, "SHA1"},
    {Digest::State::Untrusted, "RIPE-MD/160"},
    {Digest::State::Trusted, "Reserved digest"},
    {Digest::State::Trusted, "Reserved digest"},
    {Digest::State::Trusted, "Reserved digest"},
    {Digest::State::Trusted, "Reserved digest"},
+   {Digest::State::Trusted, "MD5"},
+   {Digest::State::Trusted, "SHA1"},
    {Digest::State::Trusted, "SHA256"},
    {Digest::State::Trusted, "SHA384"},
    {Digest::State::Trusted, "SHA512"},
