diff -Naur a/pyproject.toml b/pyproject.toml
--- a/pyproject.toml	2025-12-30 16:47:15.000000000 +0100
+++ b/pyproject.toml	2026-01-01 14:44:48.461597897 +0100
@@ -1,5 +1,5 @@
 [build-system]
-requires = ["setuptools >= 61.0", "setuptools-scm>=8"]
+requires = ["setuptools >= 61.0"]
 build-backend = "setuptools.build_meta"
 
 [project]
@@ -72,8 +72,6 @@
     "tools/dumppdf.py",
 ]
 
-[tool.setuptools_scm]
-
 [tool.ruff]
 target-version = "py310"
 
