diff -Naur a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	2026-06-30 14:38:31.000000000 +0200
+++ b/CMakeLists.txt	2026-07-09 16:24:18.910346314 +0200
@@ -91,7 +91,7 @@
 find_package(PlasmaWaylandProtocols REQUIRED)
 find_package(LayerShellQt REQUIRED)
 find_package(KPipeWire)
-find_package(OpenCV 4.7 REQUIRED core imgproc)
+find_package(OpenCV 5.0 REQUIRED core imgproc)
 find_package(PkgConfig REQUIRED)
 
 set_package_properties(KPipeWire PROPERTIES DESCRIPTION
