diff -Naur a/tools/power/cpupower/Makefile b/tools/power/cpupower/Makefile
--- a/tools/power/cpupower/Makefile	2025-10-15 12:00:25.000000000 +0200
+++ b/tools/power/cpupower/Makefile	2025-10-19 13:33:35.168507078 +0200
@@ -23,7 +23,7 @@
 
 # Set the following to `true' to make a unstripped, unoptimized
 # binary. Leave this set to `false' for production use.
-DEBUG ?=	true
+DEBUG ?=	false
 
 # make the build silent. Set this to something else to make it noisy again.
 V ?=		false
