diff -Naur a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	2026-06-11 09:33:15.000000000 +0200
+++ b/CMakeLists.txt	2026-06-20 21:23:52.631733140 +0200
@@ -89,7 +89,7 @@
 set(SYSTEM_CONFIG_DIR           "${CMAKE_INSTALL_PREFIX}/lib/plasmalogin/plasmalogin.conf.d"      CACHE PATH      "Path of the system plasmalogin config directory")
 set(LOG_FILE                    "${CMAKE_INSTALL_FULL_LOCALSTATEDIR}/log/plasmalogin.log"  CACHE PATH      "Path of the plasmalogin log file")
 set(DBUS_CONFIG_FILENAME        "org.freedesktop.DisplayManager.conf"               CACHE STRING    "Name of the plasmalogin config file")
-set(PLASMALOGIN_INITIAL_VT             "1"                                                 CACHE STRING    "Initial tty to use")
+set(PLASMALOGIN_INITIAL_VT             "7"                                                 CACHE STRING    "Initial tty to use")
 
 if(NOT DEFINED LOGIN_DEFS_PATH)
     set(LOGIN_DEFS_PATH "/etc/login.defs" CACHE PATH "Path to login.defs")
